core/state-machine/src/ext.rs

```diff
@@ -228,20 +235,41 @@ where
 		self.overlay.set_storage(key, value);
 	}
 
-	fn place_child_storage(&mut self, storage_key: ChildStorageKey<H>, key: Vec<u8>, value: Option<Vec<u8>>) {
+	fn place_child_storage(&mut self, child_trie: &ChildTrie, key: Vec<u8>, value: Option<Vec<u8>>) {
 		let _guard = panic_handler::AbortGuard::new(true);
 
 		self.mark_dirty();
-		self.overlay.set_child_storage(storage_key.into_owned(), key, value);
+		self.overlay.set_child_storage(child_trie, key, value);
 	}
 
-	fn kill_child_storage(&mut self, storage_key: ChildStorageKey<H>) {
+	fn set_child_trie(&mut self, ct: ChildTrie) -> bool {
+		let _guard = panic_handler::AbortGuard::new(true);
+		// do check for backend
+		let ct = match self.child_trie(ct.parent_slice()) {
+			Some(ct_old) => if
+				(ct_old.root_initial_value() != ct.root_initial_value()
```

Comment:
I don't understand this. If the `ChildTrie` is new, then it can overwrite an existing one with a different root, but if it is not new (presumably an update) then the root cannot have changed?

Reply:
Root should never be change directly like this. :+1: